test/e2e/prometheusadapter_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"log"
+	"testing"
+	"time"
+
+	"github.com/openshift/cluster-monitoring-operator/pkg/manifests"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestPrometheusAdapterCARotation(t *testing.T) {
+	// Wait for Prometheus adapter
+	err := wait.Poll(time.Second, 5*time.Minute, func() (bool, error) {
+		_, err := f.KubeClient.Apps().Deployments(f.Ns).Get("prometheus-adapter", metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	cmClient := f.KubeClient.CoreV1().ConfigMaps("kube-system")
+	cm, err := cmClient.Get("extension-apiserver-authentication", metav1.GetOptions{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Simulate rotation by simply adding a newline to existing certs.
+	// This change will be propagated to the cluster monitoring operator,
+	// causing a new secret to be created.
+	cm.Data["requestheader-client-ca-file"] = cm.Data["requestheader-client-ca-file"] + "\n"
+	cm, err = cmClient.Update(cm)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	factory := manifests.NewFactory("openshift-monitoring", nil)
+	newSecret, err := factory.PrometheusAdapterAPIAuthSecret(cm.Data)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Wait for the new secret to be created
+	err = wait.Poll(time.Second, 5*time.Minute, func() (bool, error) {
+		_, err := f.KubeClient.CoreV1().Secrets(f.Ns).Get(newSecret.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
```

Comment:
Same as above, let's surface this error to the user, see: https://github.com/openshift/cluster-monitoring-operator/pull/240/files